### PR TITLE
implement tracking and fix slack message

### PR DESCRIPTION
**Pull request by ImprobableVlad**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR adds automatic tracking of some GDK test-related metrics. These metrics are uploaded to BigQuery and are displayed in the following DataStudio dashboard: https://brevi.link/gdk-for-unreal-testing-metrics. Metrics are only uploaded when NIGHTLY_BUILD is set to true.

#### Release note
This is not a public-facing change, so no release note is included.

#### Tests
A build was run with the NIGHTLY_BUILD environment variable set to _true_. The results was indeed uploaded to BigQuery, and can be seen on the dashboard: https://brevi.link/gdk-for-unreal-testing-metrics

#### Documentation
In-code documentation has been included.

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
@mironec @jamiebrynes7 @michelefabris

### CODE REVIEW

File: ci/report-tests.ps1 [hunk 1]
```diff
@@ -75,7 +84,7 @@ $slack_attachment = [ordered]@{
                 short = $true
             }
             @{
-                value = "Passed $($test_results_obj.succeeded) / $($test_results_obj.succeeded + $test_results_obj.failed) tests."
+                value = "Passed $($test_results_obj.succeeded + $test_results_obj.succeededWithWarnings) / $num_gdk_tests tests."
```
Comment (ImprobableVlad) [inline ci/report-tests.ps1]: This line fixes a minor bug where tests that succeeded with warnings were not reported as being successfull in the slack message. 
image

File: ci/upload-test-metrics.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    --write-to=${GCP_SECRET}
+gcloud auth activate-service-account --key-file "${GCP_SECRET}"
+
+# Define target upload location
+PROJECT="holocentroid-aimful-6523579"
+DATASET="UnrealGDK"
+TABLE="ci_metrics"
+
+for json_file in ./test_summaries/*.json; do
+    cat $json_file | bq --project_id "$PROJECT" --dataset_id "$DATASET" insert "$TABLE"
```
Comment (jamiebrynes7) [inline ci/upload-test-metrics.sh]:
```suggestion
    cat "${json_file}" | bq --project_id "${PROJECT}" --dataset_id "{$DATASET}" insert "${TABLE}"
```

Comment (jamiebrynes7) [reply ci/upload-test-metrics.sh]: For consistency + to handle the case where `json_file` has spaces

File: ci/report-tests.ps1 [hunk 3]
```diff
@@ -99,6 +108,21 @@ $slack_attachment | ConvertTo-Json | Set-Content -Path "$test_result_dir\slack_a
...
+    platform = "$target_platform"
+    unreal_engine_commit = "$env:ENGINE_COMMIT_HASH"
+    passed_all_tests = $tests_passed
+    tests_duration_seconds = $test_results_obj.totalDuration
+    num_tests = $test_results_obj.succeeded + $test_results_obj.failed
+    num_gdk_tests = $num_gdk_tests
+}
+$test_summary | ConvertTo-Json -Compress | Set-Content -Path "$test_result_dir\test_summary_$env:BUILDKITE_STEP_ID.json"
+
+buildkite-agent "artifact" "upload" "$test_result_dir\test_summary_$env:BUILDKITE_STEP_ID.json"
```
[the remark this answers is not in the document]
Comment (ImprobableVlad) [reply ci/report-tests.ps1]: Agreed - fixed it now

File: ci/upload-test-metrics.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -euo pipefail
+
+# Fetch the test summary artifacts uploaded earlier
+mkdir "./test_summaries"
+buildkite-agent artifact download "*test_summary*.json" "./test_summaries"
+
+# Fetch Google credentials so that we can upload the metrics to the GCS bucket.
+GCP_SECRET="$(mktemp)"
```
[the remark this answers is not in the document]
Comment (ImprobableVlad) [reply ci/upload-test-metrics.sh]: Implemented it now. Does it look good to you?

File: ci/upload-test-metrics.sh (hunk 2 above)
Comment (ImprobableVlad) [reply ci/upload-test-metrics.sh]: Yup, good one - fixed it now.

File: ci/report-tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (ImprobableVlad) [reply ci/report-tests.ps1]: Absolutely! I've rewritten this part to work properly.

File: ci/upload-test-metrics.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+# Define target upload location
+PROJECT="holocentroid-aimful-6523579"
+DATASET="UnrealGDK"
+TABLE="ci_metrics"
+
+# Make sure that the gcp secret is always removed
+GCP_SECRET="$(mktemp)"
+function cleanup {
+  rm -rf "$GCP_SECRET"
```
Comment (jamiebrynes7) [inline ci/upload-test-metrics.sh]:
```suggestion
  rm -rf "${GCP_SECRET}"
```

File: ci/upload-test-metrics.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply ci/upload-test-metrics.sh]: Yup - looks good! :)

**Pull request merged**